proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+```
+
+Currently, it is required to cover every reference to a `spawn let` using the apropriate try and await keywords, like this:
+
+```swift
+spawn let yes = ""
+spawn let ohNo = throwThings()
+
+_ = await yes
+_ = await yes
+_ = try await ohNo
+_ = try await ohNo
+```
+
+This is a simple rule and allows us to bring the feature forward already. It might be possible to employ control flow based analysis to enable "only the first reference to the specific `spawn let` on each control flow path has to be an `await`", as technically speaking, every following await will be a no-op and will not suspend as the value is already completed and the placeholder has been filled in.
+
+### Implicit `spawn let` awaiting 
+
+A `spawn let` that was declared but never awaited on *explicitly* as the scope in which it was declared exits, will be awaited on implicitly. These semantics are put in place to uphold the Structured Concurrency guarantees provided by `spawn let`.
+
+To showcase these semantics, let us have a look at this function which spawns two child tasks, `fast` and `slow` but does not await on any of them:
+
+```swift
+func go() async { 
+  spawn let f = fast() // 300ms
+  spawn let s = slow() // 3seconds
+  return "nevermind..."
+  // implicitly: cancels f
+  // implicitly: cancels s
+  // implicitly: await f
+  // implicitly: await s
+}
+```
+
+Assuming the execution times of fast and slow are as the comments next to them explain, the `go()` function will _always_ take at least 3 seconds to execute. Or to state the rule more generally, any structured invocation, will take as much time to return as much the longest of it child tasks takes to complete.
+
+As we return from the `go()` function without ever having awaited on the `f` or `s` values, both of them will be implicitly cancelled and awaited on before returning from the function `go()`. This is the very nature of structured concurrency, and avoiding this can _only_ be done by creating non-child tasks, e.g. by using `detach` or other future APIs which would allow creation of non-child tasks.
+
+If we instead awaited on one of the values, e.g. the fast one (`f`) the emitted code would not need to implicitly cancel or await it, as this was already taken care of explicitly:
+
+```swift
+func go2() async {
+  spawn let f = fast()
+  spawn let s = slow()
+  _ = await f
+  return "nevermind..."
+  // implicitly: cancels s
+  // implicitly: awaits s
+}
+```
+
+The duration of the `go2()` call remains the same, it is always `time(go2) == max(time(f), time(s))`.
+
+Special attention needs to be given to the `spawn let _ = ...` form of spawn let declarations. This form of is interesting because it creates a child-task of the right hand-side initializer, however actively chooses to ignore the result. Such declaration, and the associated child-task will run, and be awaited-on implicitly as the scope it was declared is about to exit - the same way as an un-used spawn let declaration would be.
```

Comment:
```suggestion
Special attention needs to be given to the `spawn let _ = ...` form of declarations. This form is interesting because it creates a child-task of the right-hand-side initializer, however it actively chooses to ignore the result. Such a declaration (and the associated child-task) will run and be awaited-on implicitly, as the scope it was declared in is about to exit — the same way as an unused `spawn let` declaration would be.
```